### PR TITLE
Infra: Use GitHub issue forms

### PR DESCRIPTION
**Pull request by JacobCoffee**
- Updates issue creation to use [Issue Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)

----
📚 Documentation preview 📚: https://cpython-devguide--1414.org.readthedocs.build/

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+body:
+  - type: markdown
+    attributes:
+      value: |
+        > [!NOTE]
+        > This repo is for the Python devguide. If you are reporting a bug
+        > for the Python language or CPython interpreter,
+        > then the CPython issue tracker is better
+        > suited for this report: https://github.com/python/cpython/issues
```
Comment (ezio-melotti) [inline .github/ISSUE_TEMPLATE/bug_report.yml:15]:
```suggestion
        > This repo is for the [Python devguide](https://devguide.python.org/).
        > If you are reporting a bug for the Python language or
        > CPython interpreter, then use the
        > [CPython issue tracker](https://github.com/python/cpython/issues) instead.
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+body:
+  - type: markdown
+    attributes:
+      value: |
+        > [!NOTE]
+        > This repo is for the Python devguide. If you are requesting an
+        > enhancement for the Python language or CPython interpreter,
+        > then the CPython issue tracker is better
+        > suited for this report: https://github.com/python/cpython/issues
```
Comment (ezio-melotti) [inline .github/ISSUE_TEMPLATE/feature_request.yml:15]:
```suggestion
        > This repo is for the [Python devguide](https://devguide.python.org/).
        > If you are requesting an enhancement for the Python language or
        > CPython interpreter, then use the
        > [CPython issue tracker](https://github.com/python/cpython/issues) instead.
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  - type: textarea
+    id: feature_description
+    attributes:
+      label: "Describe the enhancement or feature you'd like"
```
Comment (ezio-melotti) [inline .github/ISSUE_TEMPLATE/feature_request.yml:20]:
```suggestion
      label: "Describe the enhancement or feature you would like"
```
Abbreviations should be avoided.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      description: A clear and concise description of what you want to happen.
+    validations:
+      required: true
+
+  - type: textarea
+    id: alternatives
+    attributes:
+      label: "Describe alternatives you've considered"
+      description: A clear and concise description of any alternative solutions or features you've considered.
```
Comment (ezio-melotti) [inline .github/ISSUE_TEMPLATE/feature_request.yml:29]:
```suggestion
      label: "Describe alternatives you have considered"
      description: A clear and concise description of any alternative solutions or features you have considered.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    id: expected_behavior
+    attributes:
+      label: "Expected behavior"
+      description: A clear and concise description of what you expected to happen.
+    validations:
+      required: true
```
Comment (AA-Turner) [inline .github/ISSUE_TEMPLATE/bug_report.yml:31]: I'm unsure how much "expected behaviour" makes sense for a documentation site. Perhaps just alter the description field and ask the user to describe the problem and what would resovle it?

Comment (JacobCoffee) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: [c634d1e](https://github.com/python/devguide/pull/1414/commits/c634d1eee3c7593682a6aff497fa3c257571ec00)

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+title: "Feature: <title>"
+labels: ["enhancement"]
+assignees: []
+
+body:
+  - type: markdown
+    attributes:
+      value: |
+        > [!NOTE]
+        > This repo is for the [Python devguide](https://devguide.python.org/).
```
Comment (AA-Turner) [inline .github/ISSUE_TEMPLATE/feature_request.yml:12]:
```suggestion
        > This repo is for the [Python developer's guide](https://devguide.python.org/).
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+title: "Bug: <title>"
+labels: ["bug"]
+assignees: []
+
+body:
+  - type: markdown
+    attributes:
+      value: |
+        > [!NOTE]
+        > This repo is for the [Python devguide](https://devguide.python.org/).
```
Comment (AA-Turner) [inline .github/ISSUE_TEMPLATE/bug_report.yml:12]:
```suggestion
        > This repo is for the [Python developer's guide](https://devguide.python.org/).
```

**Review by AA-Turner — APPROVED** [review]

**Review by willingc — APPROVED** [review]

**Pull request merged**